util/feature/manager.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 PingCAP, Inc. Licensed under Apache-2.0.
+
+package feature
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+
+	"github.com/gin-gonic/gin"
+)
+
+type Manager struct {
+	version      string
+	flags        []*Flag
+	supportedMap map[string]struct{}
+}
+
+func NewManager(version string) *Manager {
+	return &Manager{
+		version:      version,
+		flags:        []*Flag{},
+		supportedMap: map[string]struct{}{},
+	}
+}
+
+// NewManagerProvider returns a fx.Provider.
+func NewManagerProvider(version string) func() *Manager {
+	return func() *Manager {
+		return NewManager(version)
+	}
+}
```

Comment:
I guess this doesn't reduce LOC as you expect, while increasing the difficulty to understand this code. For example, without this function, in order to use the FeatureManager in fx, it is not that hard:

```go
fx.Provide(
   ...
   feature.NewManager(version)),
   ...
)
```

While in the current code base, you forces the reader to understand what is a "Provider". There are also many different "Provider" concepts in this repo, which will confuse the readers deeper.

Reply:
It looks like we should provide a constructor function for `fx.Provide`.
https://pkg.go.dev/go.uber.org/fx#Provide